code/src/nuvla/job/actions/deployment_start.py

```diff
@@ -4,15 +4,24 @@
 
 from nuvla.api.resources import Deployment, DeploymentParameter
 from nuvla.api.resources.base import ResourceNotFound
-from nuvla.connector import connector_factory, docker_connector, \
-    docker_cli_connector, docker_compose_cli_connector, kubernetes_cli_connector
+from nuvla.connector import docker_connector, docker_cli_connector, \
+    docker_compose_cli_connector, kubernetes_cli_connector
 from ..actions import action
 
 action_name = 'start_deployment'
 
 log = logging.getLogger(action_name)
 
 
+def initialize_connector(connector_class, job, deployment):
+    credential_id = Deployment.credential_id(deployment)
+    credential = job.context[credential_id]
+    infrastructure_service = job.context[credential['parent']]
+    if job.is_in_pull_mode:
```

Comment:
isn't this also needed for deployment_stop/update/state?

Reply:
Yes, this is why it's imported into all of them